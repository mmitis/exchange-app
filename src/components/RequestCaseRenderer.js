//@flow
import React from "react";
import type {RequestStatus} from "../constants/Types";
import RequestStatuses from "../constants/RequestStatuses";

type Props = {
    status: RequestStatus,
    error: string,
    children: React$Node
}
export const RequestCaseRenderer = ({
                                  status,
                                  error,
                                  children
} : Props) => {
    switch(status) {
        case RequestStatuses.NOT_PENDING:
            return null;
        case RequestStatuses.IN_PROGRESS:
            return <div className="spinner" />;
        case RequestStatuses.SUCCESS:
            return children;
        case RequestStatuses.FAILURE:
            return <div className="error-message">{error}</div>;
        default:
            return null;
    }
};

export default RequestCaseRenderer;
