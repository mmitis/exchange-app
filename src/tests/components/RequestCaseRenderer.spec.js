import RequestCaseRenderer from './../../components/RequestCaseRenderer';
import { mount } from 'enzyme';
import React from "react";

describe('Component: RequestCaseRenderer', () => {

    it('test simple render - in_progress', async () => {
        const props = {
           status: 'IN_PROGRESS'
        };
        const rotate = mount(<RequestCaseRenderer
            {...props}
        >
            <div className={'sample-div'}></div>

        </RequestCaseRenderer>);
        
        expect(rotate).toMatchSnapshot();
    });

    it('test simple render - error', async () => {
        const props = {
           status: 'FAILURE',
           error: 'Sample Error'
        };
        const rotate = mount(<RequestCaseRenderer
            {...props}
        >
            <div className={'sample-div'}></div>

        </RequestCaseRenderer>);
        
        expect(rotate).toMatchSnapshot();
    });

    it('test simple render - success', async () => {
        const props = {
           status: 'SUCCESS'
        };
        const rotate = mount(<RequestCaseRenderer
            {...props}
        >
            <div className={'sample-div'}></div>

        </RequestCaseRenderer>);
        
        expect(rotate).toMatchSnapshot();
    });

});
