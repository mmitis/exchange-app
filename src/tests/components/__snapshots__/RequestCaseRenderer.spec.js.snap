// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RequestCaseRenderer test simple render - error 1`] = `ReactWrapper {}`;

exports[`Component: RequestCaseRenderer test simple render - in_progress 1`] = `ReactWrapper {}`;

exports[`Component: RequestCaseRenderer test simple render - success 1`] = `ReactWrapper {}`;
